### PR TITLE
docs: updated docs of the library

### PR DESCRIPTION
**Pull request by ghost**
This PR updates the docs of this library.

Some RFC urls are removed from reference because Go doc automatically renders the RFC as a hyperlink. For example, `RFC 3161` is rendered as a hyperlink pointing to the RFC 3161 doc online.

### CODE REVIEW

File: tspclient.go [hunk 1]
```diff
@@ -13,8 +13,7 @@
 
 // Package tspclient generates timestamping requests to TSA servers,
 // fetches and verifies the responses according to
-// RFC 3161: https://datatracker.ietf.org/doc/html/rfc3161 &
-// RFC 5816: https://datatracker.ietf.org/doc/html/rfc5816
+// RFC 3161 & RFC 5816
```
Comment (shizhMSFT) [inline tspclient.go]: replace `&` with `and`? You may also merge line 15 with line 16.

File: README.md
Comment (shizhMSFT) [inline README.md:1]: Can you also fill in other sections like documentation, code of conduct and license? See https://github.com/notaryproject/notation-core-go/blob/main/README.md

Comment (ghost) [reply README.md:1]: Added.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]

**Pull request merged**